File: backend/src/TrailWise.Domain/Entities/Review.cs
namespace TrailWise.Domain.Entities;

public class Review : BaseEntity
{
    public Guid BookingId { get; set; }
    public Booking Booking { get; set; } = null!;

    public int Rating { get; set; }
    public string? Comment { get; set; }
    public DateTimeOffset SubmittedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: frontend-mobile/lib/models/review.dart
class Review {
  final String id;
  final String bookingId;
  final int rating;
  final String? comment;
  final String submittedAt;

  Review({
    required this.id,
    required this.bookingId,
    required this.rating,
    this.comment,
    required this.submittedAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) => Review(
        id: json['id'] as String,
        bookingId: json['bookingId'] as String,
        rating: (json['rating'] as num).toInt(),
        comment: json['comment'] as String?,
        submittedAt: json['submittedAt'] as String,
      );
}
